src/ray/raylet/scheduling_resources.cc

```diff
@@ -73,19 +73,19 @@ bool ResourceSet::RemoveResource(const std::string &resource_name) {
   throw std::runtime_error("Method not implemented");
 }
 bool ResourceSet::SubtractResources(const ResourceSet &other) {
-  // Return failure if attempting to perform vector subtraction with unknown labels.
-  // TODO(atumanov): make the implementation atomic. Currently, if false is returned
-  // the resource capacity may be partially mutated. To reverse, call AddResources.
+  // Subtract the resources and track whether a resource goes below zero.
+  bool oversubscribed = false;
   for (const auto &resource_pair : other.GetResourceMap()) {
     const std::string &resource_label = resource_pair.first;
     const double &resource_capacity = resource_pair.second;
-    if (resource_capacity_.count(resource_label) == 0) {
-      return false;
-    } else {
-      resource_capacity_[resource_label] -= resource_capacity;
+    // TODO(atumanov): Return failure if attempting to perform vector
+    // subtraction with unknown labels.
+    resource_capacity_[resource_label] -= resource_capacity;
```

Comment:
We need to handle the case where `resource_capacity_.count(resource_label) == 0)` (that is the local scheduler is not aware of this resource). Or is that case impossible?

Reply:
Hmm yeah, it seemed like Alexey had left that as a todo, but I can just put a check here for now. I think we should just do a fatal check for that for now...